cdap-common/src/main/java/co/cask/cdap/common/security/Impersonator.java

```diff
@@ -62,6 +63,7 @@
    * @param namespaceId namespace to lookup the user
    * @return {@link UserGroupInformation}
    * @throws IOException if there was any error fetching the {@link UserGroupInformation}
+   * @throws NamespaceNotFoundException if namespaceId does not exist
    */
-  UserGroupInformation getUGI(NamespaceId namespaceId) throws IOException;
+  UserGroupInformation getUGI(NamespaceId namespaceId) throws IOException, NamespaceNotFoundException;
```

Comment:
Done. Since `doAs()` throws Exception, I have added @throws in doAs(namespaceId, callable).  That is the only place where we can get `NamespaceNotFoundException`.